service/role/service.go

```diff
@@ -0,0 +1,114 @@
+package role
+
+import (
+	"fmt"
+
+	"github.com/giantswarm/microerror"
+	vaultclient "github.com/hashicorp/vault/api"
+)
+
+// Config defines configurable aspects (such as dependencies) of this service.
+type Config struct {
+	// Config
+	PKIMountpoint string
+
+	// Dependencies.
+	VaultClient *vaultclient.Client
+}
+
+// DefaultConfig returns a default configuration that can be used to create this service.
+func DefaultConfig() Config {
+	vaultClientConfig := vaultclient.DefaultConfig()
+	vaultClientConfig.Address = "http://127.0.0.1:8200"
+	vaultClient, err := vaultclient.NewClient(vaultClientConfig)
+	if err != nil {
+		panic(err)
+	}
+
+	config := Config{
+		// Dependencies.
+		VaultClient: vaultClient,
+	}
+
+	return config
+}
+
+// New takes a configuration and returns a configured service.
+func New(config Config) (Service, error) {
+	// Dependencies.
+	if config.VaultClient == nil {
+		return nil, microerror.Maskf(invalidConfigError, "Vault client must not be empty")
+	}
+
+	if config.PKIMountpoint == "" {
+		return nil, microerror.Maskf(invalidConfigError, "PKIMountpoint must not be empty")
+	}
+
+	service := &service{
+		Config: config,
+	}
+
+	return service, nil
+}
+
+type service struct {
+	Config
```

Comment:
FYI it would be good to assign the things in the config as private properties here. This prevents the user from changing things he should not change. It is just a better design. Not critical at this point though but we should try to copy/paste the right code.